tests/lib/rules/no-unnecessary-get-properties.js

```diff
@@ -0,0 +1,108 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-unnecessary-get-properties');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE } = rule;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    sourceType: 'module'
+  }
+});
+
+ruleTester.run('no-unnecessary-get-properties', rule, {
+  valid: [
+    // Simple variable declaration.
+    'const abc = 123;',
+
+    // Destructuring assignment but no `getProperties`.
+    'const { abc, def } = myObject;',
+
+    // `myObject` variable but no destructuring.
+    "const properties = myObject.getProperties('abc', 'def');",
+    "const properties = myObject.getProperties(['abc', 'def']);", // With parameters in array.
+    "const properties = getProperties(myObject, 'abc', 'def');",
+    "const properties = getProperties(myObject, ['abc', 'def']);", // With parameters in array.
+
+    // `this` but no destructuring.
+    "const properties = this.getProperties('abc', 'def');",
+    "const properties = this.getProperties(['abc', 'def']);", // With parameters in array.
+    "const properties = getProperties(this, 'abc', 'def');",
+    "const properties = getProperties(this, ['abc', 'def']);", // With parameters in array.
+
+    // Destructuring assignment with nested path.
+    "const { foo, barBaz } = myObject.getProperties('foo', 'bar.baz');",
+    "const { foo, barBaz } = myObject.getProperties(['foo', 'bar.baz']);", // With parameters in array.
+    "const { foo, barBaz } = getProperties(myObject, 'foo', 'bar.baz');",
+    "const { foo, barBaz } = getProperties(myObject, ['foo', 'bar.baz']);", // With parameters in array.
+
+    // With some wrapping function call.
+    "const { abc, def } = someFunction(myObject.getProperties('abc', 'def'));",
+
+    // With non-string parameter.
+    'const { abc } = this.getProperties(MY_PROP);',
+    'const { abc } = this.getProperties([MY_PROP]);',
+    'const { abc } = getProperties(this, MY_PROP);',
+    'const { abc } = getProperties(this, [MY_PROP]);'
+  ],
+  invalid: [
```

Comment:
can you add a test case for e.g. `this.getProperties('while')`? this should not convert to `let { while } = ...` since that would be illegal. but then again, this rule only triggers when there was already destructuring used before, right?

(see also https://github.com/rondale-sc/es5-getter-ember-codemod/blob/6e7929832301485ff15c0f4384363a1040eabb5c/transforms/es5-getter-ember-codemod/index.js#L3-L7)

Reply:
Interesting! However, I don't think that applies here, because this rule is only suggesting removing the `getProperties` call, and not changing the existing destructuring.